core/src/main/java/org/elasticsearch/http/netty/NettyHttpChannel.java

```diff
@@ -134,6 +137,7 @@ public void sendResponse(RestResponse response) {
             if (!addedReleaseListener && content instanceof Releasable) {
                 ((Releasable) content).close();
             }
+            transport.inFlightRequestsBreaker.addWithoutBreaking(-reservedBytes);
```

Comment:
can we make this implement AutoClosable, call the close method from here to do all the clean up and make sure it's only closed once?